lib/rules/blank-lines.js

```diff
@@ -19,30 +32,85 @@ module.exports = {
         docs: {
             recommended: true,
             type: "warning",
-            description: "Ensure that there is exactly a 2-line gap between Contract and Funtion declarations"
+            description: "Ensure that there is exactly a 2-line gap between Contract and Function declarations"
         },
 
+        fixable: "whitespace",
+
         schema: []
 
     },
 
     create: function(context) {
 
         let sourceCode = context.getSourceCode(),
-            noCodeRegExp = /^[ \t]*$/,
-            isCommentRegExp = /\s*(\/\/*)|(\*\\*)|(\/\**)$/;
+            noCodeRegExp = /^\s*$/,
+            isCommentRegExp = /\s*(\/\/*)|(\*\\*)|(\/\**)$/,
+            strictCommentRegExp = /^\s*(\/\/*)|(\*\\*)|(\/\**)$/; // Tests whether the entire line is a comment
         let topLevelDeclarations = [ "ContractStatement", "LibraryStatement" ];
 
         function inspectProgram(emitted) {
             if (emitted.exit) {
                 return;
             }
 
-            function report(topLevelNode) {
+            function report(topLevelNode, spacing) {
                 const message = `${topLevelNode.type.replace("Statement", "")} '${topLevelNode.name}' `
                     + "must be preceded by 2 blank lines.";
 
-                context.report({ node: topLevelNode, message });
+                context.report({
+                    node: topLevelNode, message,
+                    fix(fixer) {
+                        // The fix strategy is as follows:
+                        // - count lines going upwards (without going beyond the bounds of the spacing between this node and the previous one)
+                        // - Skip over any accompanying comments of the top level node
+                        // - Count the number of blank lines directly above the top-most accompanying comments iteratively
+                        // - Insert the desired number of line break characters right before the comment
+
+                        let line = sourceCode.getLine(topLevelNode) - 1, // The line number
+                            maxLines = spacing.split("\n").length,
```

Comment:
Are you sure we don't need to use `EOL` instead of just `\n` over here? And in all the places where you're using `\n` only?

Reply:
The code would be equivalent either way since `\r\n` includes `\n`. Although you could argue that splitting by `\n` alone causes inconsistent behavior across OSes since on windows there would be an extra `\r` in the substrings resulting from the split. In the context of counting the number of lines it's not really a big deal if the strings have that extra `\r`. If you'd prefer to use `EOL` for the sake of consistency however that's a good reason and I could definitely change it up.

Reply:
Let's use `EOL`
1. for consistency 
2. so there's no scope of confusion about what kind of linebreak to use where for those who read this rule's code in future

Reply:
Sounds good. My latest commit should cover it.